### ISSUE TITLE
Unblocked sites available to check in lists cycle

### ISSUE THREAD

**Issue by JimKillock** [openrightsgroup/blocked-org-uk#233]
When cycling through a list of sites, those sites that are unblocked appear in the selection of sites to check / unblock.

**Comment by JimKillock** [comment]
May be related to #189

**Comment by dantheta** [comment]
There have been a lot of changes to this today.  The next time you see one, can you post it as an example?

**Comment by JimKillock** [comment]
Sure @dantheta, here's an example listed from the front page: https://www.blocked.org.uk/site/http://www.imtasik.org

**Comment by JimKillock** [comment]
I tried using the Cardiff list and got: https://www.blocked.org.uk/site/http://www.ukdrivercpcnetwork.co.uk in the cycle

**Comment by dantheta** [comment]
Both interesting cases - imtasik was updated from the backend DB at midnight, and resubmitted for testing at 04:30.  It would pick up the unblocked status some time in the next 24 hours and stop appearing in the frontpage results then.  

I'm going to move the storage for the saved lists into the backend DB where it can be joined against the latest results every time it's called.  We won't get secondhand data any more.

**Comment by dantheta** [comment]
Found & fixed another place where they were creeping in.  The random selection is quite a complicated workflow ...

**Comment by dantheta** [comment]
There's still a chance that the odd unblocked site may creep in, but only in cases where it was unblocked within the last day.   I'll spin the list move into a separate issue.
